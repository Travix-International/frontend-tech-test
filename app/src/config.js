export const config = {
  api: 'http://localhost:4000'
};

export default config;
