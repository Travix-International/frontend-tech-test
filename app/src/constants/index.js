export const TASK_REQUEST = 'TASK_REQUEST';
export const TASK_LIST = 'TASK_LIST';
export const TASK_SAVE = 'TASK_SAVE';
export const TASK_DELETE = 'TASK_DELETE';
export const NOTIFICATION = 'NOTIFICATION';
