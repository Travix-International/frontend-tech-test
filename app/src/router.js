import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

// import { history } from './store';
import Main from './modules/main/component';
import TaskContainer from './modules/taskRegister/container';

const Routes = () => (
  <Router /* history={history} */>
    <Switch>
      <Route path="/task/:id" component={TaskContainer} />
      <Route component={Main} />
    </Switch>
  </Router>
);

export default Routes;
