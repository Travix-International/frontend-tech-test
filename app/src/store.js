import thunk from 'redux-thunk';
import { createStore, compose, applyMiddleware } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';

import reducersCombined from './modules/combinedReducers';

// If you have a Redux extesion for Chrome.
const enhacers = (window.devToolsExtension ? window.devToolsExtension() : f => f);

export const history = createHistory();

export const store = createStore(
  reducersCombined,
  undefined,
  compose(
    applyMiddleware(thunk, routerMiddleware(history)),
    enhacers
  )
);
