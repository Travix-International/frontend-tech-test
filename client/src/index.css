body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background: #0b0e19 url(./logo.svg);
  background-size: 150px;
}

* {
  box-sizing: border-box;
}

ul {
  padding: 0;
  margin: 0;
}

input[type=text] {
  width: 100%;
  max-width: 100%;
  padding: 1rem;
  font-size: 1rem;
  margin: 0 auto;
  border-radius: 4px;
  border-width: 0;
}

input[type=text] + input[type=text] {
  margin-top: 1rem;
}

input[type=submit] {
  display: none;
}

input[type=text]#inputTodoTitle {
  font-size: 1.5rem;
  font-weight: bold;
}

input[type=text]#inputTodoDescription {
  height: 5rem;
}

.container {
  max-width: 800px;
  margin: 0 auto;
}

header {
  background-color: #2196F3;
  color: #FFF;
  padding: 3rem 0;
  margin-bottom: 1rem;
}

ol.todos {
  margin: 0;
  padding: 0;
  list-style: none;
}

ol.todos li {
  padding: 0.5rem 0;
  background: rgba(255,255,255,0.95);
  padding: 1rem;
  margin-bottom: .5rem;
  border-radius: 4px;
  border-width: 0;
}
