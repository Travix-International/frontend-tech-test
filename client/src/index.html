<!DOCTYPE html>
<html>

<head>
  <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
  <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" rel="stylesheet">
  <title>Test Project</title>
</head>

<body>
  <div id="root"></div>
</body>

</html>