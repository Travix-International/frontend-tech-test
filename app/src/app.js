import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { store } from './store';
import Routes from './router';

const App = () => (
  <Provider store={store}>
    <Routes />
  </Provider>
);

const root = document.getElementById('root');

render(<App />, root);
