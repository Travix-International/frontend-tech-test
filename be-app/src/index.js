import { createServer } from 'http';
import app from './server';

const server = createServer(app);
let currentApp = app;

server.listen(9001, () => {
  console.log('Server listening on port 9001');
});

if (module.hot) {
  module.hot.accept(['./server'], () => {
    server.removeListener('request', currentApp);
    server.on('request', app);
    currentApp = app;
  });
}

