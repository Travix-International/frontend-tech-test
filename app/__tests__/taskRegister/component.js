import React from 'react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TaskRegister from './../../src/modules/taskRegister/component';

import { history } from './../../src/store';

describe('TaskRegister Tests', () => {
  test('TaskRegister Render', () => {
    const props = {
      task: {
        _id: '',
        title: '',
        description: '',
        date: '',
        completed: false
      },
      taskSave: () => {}
    };

    const tree = renderer.create(
      <Router history={history}>
        <TaskRegister {...props} />
      </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
