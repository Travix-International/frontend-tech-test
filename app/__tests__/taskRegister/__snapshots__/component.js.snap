exports[`TaskRegister Tests TaskRegister Render 1`] = `
<form
  className="task__form"
  id="taskForm"
  onSubmit={[Function]}>
  <div
    className="task__form__content">
    <h1>
      Register Task
    </h1>
    <span
      className="">
      Title
    </span>
    <input
      className=""
      defaultValue=""
      required={true}
      type="text" />
    <span
      className="">
      Description
    </span>
    <input
      className=""
      defaultValue=""
      required={true}
      type="text" />
    <span
      className="">
      Date
    </span>
    <input
      className=""
      defaultValue="Invalid date"
      required={true}
      type="date" />
    <div>
      <input
        className=""
        defaultChecked={false}
        id="register-task-completed"
        type="checkbox" />
      <label
        htmlFor="register-task-completed">
        Completed
      </label>
    </div>
  </div>
  <div
    className="task__form__buttons">
    <button
      className="btn"
      type="submit">
      Send
    </button>
    <a
      className="btn"
      href="/"
      onClick={[Function]}>
      Back
    </a>
  </div>
</form>
`;
